### PR TITLE
Compiler Enum Support and updated flyteidl #minor

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
Support for ENUM types in flytecompiler and flytepropeller update to support it

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
https://docs.google.com/document/d/1rf0hMKbPebJ7jGYpsTDAbJ4ka58KIKD7EgWAEBIH06Q/edit#

## Tracking Issue
https://github.com/flyteorg/flyte/issues/590

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/compiler/validators/typing.go [hunk 1]
```diff
@@ -33,6 +33,20 @@ func (t trivialChecker) CastsFrom(upstreamType *flyte.LiteralType) bool {
...
 		return true
 	}
+
+	// If upstream is an enum, it can be consumed as a string downstream
+	if upstreamType.GetEnumType() != nil {
+		if t.literalType.GetSimple() == flyte.SimpleType_STRING {
+			return true
+		}
+	}
+	// If t is an enum, it can be created from a string as Enums as just constrained String aliases
```
Comment (katrogan) [inline pkg/compiler/validators/typing.go:43]: is this a runtime error then if the string isn't a valid enum value?

**Review by katrogan — APPROVED** [review]

File: pkg/compiler/validators/typing.go (hunk 1 above)
Comment (kumare3) [reply pkg/compiler/validators/typing.go:43]: yup

Comment (kumare3) [reply pkg/compiler/validators/typing.go:43]: but our entry points will not allow bad values

Comment (kumare3) [reply pkg/compiler/validators/typing.go:43]: @katrogan thats actually a good point. Can we restrict this in flyteadmin? Where we construct the launchplan?

Comment (katrogan) [reply pkg/compiler/validators/typing.go:43]: shouldn't the launch plan have to reference the enum type to be valid?

File: pkg/compiler/validators/typing.go [hunk 2]
```diff
@@ -33,6 +33,20 @@ func (t trivialChecker) CastsFrom(upstreamType *flyte.LiteralType) bool {
 	if isVoid(upstreamType) {
 		return true
 	}
+
+	// If upstream is an enum, it can be consumed as a string downstream
+	if upstreamType.GetEnumType() != nil {
```
Comment (EngHabu) [inline pkg/compiler/validators/typing.go:38]: We should also add a check to the static binding to fail if the string isn't in the list...

**Pull request merged**